feat: add `math/base/special/acoshf`

0PrashantYadav0 opened the pull request:
Resolves None.

## Description

> What is the purpose of this pull request?

This pull request:

-   added base implementation for `math/base/special/acoshf`
-   added benchmark, example and test.
-   added docs and C-implementation.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   ref #649

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

0PrashantYadav0 committed:
feat: add math/special/acoshf

0PrashantYadav0 commented:
Depend on #5803

0PrashantYadav0 committed:
chore: add test for acoshf

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented:
<!-- stdlib-bot-tracking-issue-closure-check -->
:warning: **Tracking Issue Closure Warning** :warning:

I noticed your PR description contains closing keywords ("Resolves", "Closes", or "Fixes") referencing a "Tracking Issue".

**Why this matters:**
Tracking issues should typically remain open until all related sub-issues are completed. GitHub automatically closes issues with such closing keywords when the PR is merged.

**Required action:**
Use "Progresses:" or "Ref:" instead to reference the tracking issue without automatically closing it.

Thank you for your contribution to the project!

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into math-acoshf

kgryte commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

nakul-krishnakumar committed:
chore: clean up cppcheck-suppress comment

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
chore: remove scripts from directory list

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
chore: add f32 conversions for computation statements

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar commented on lib/node_modules/@stdlib/math/base/special/acoshf/test/test.native.js:
```diff
@@ -0,0 +1,149 @@
+			t.equal( y, e, 'x: '+x[i]+'. E: '+e );
+		} else {
+			delta = abs( y - e );
+			tol = 1.0 * EPS * abs( e );
```
```suggestion
			delta = absf( y - e );
			tol = 1.0 * EPS * absf( e );
```

nakul-krishnakumar commented on lib/node_modules/@stdlib/math/base/special/acoshf/test/test.native.js:
```diff
@@ -0,0 +1,149 @@
+			t.equal( y, e, 'x: '+x[i]+'. E: '+e );
+		} else {
+			delta = abs( y - e );
+			tol = 1.0 * EPS * abs( e );
```
```suggestion
			delta = absf( y - e );
			tol = 1.0 * EPS * absf( e );
```

nakul-krishnakumar committed:
chore: add f64 to f32 conversions

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
chore: update according to code review

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/acoshf/benchmark/benchmark.js:
```diff
@@ -0,0 +1,81 @@
+
+var opts = {
+	'skip': ( typeof Math.acoshf !== 'function' ) // eslint-disable-line stdlib/no-builtin-math
+};
```
This is not needed. JavaScript does not have `Math.acoshf`

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/acoshf/benchmark/benchmark.js:
```diff
@@ -0,0 +1,81 @@
+	var y;
+	var i;
+
+	x = uniform( 100, 1.0, 100.0 );
```
Missing `opts` argument.

Should be changed to:

```js
    x = uniform( 100, 1.0, 100.0, {
        'dtype': 'float32'
    });
```

anandkaranubc replied:
Same for `benchmark.native.js`

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/acoshf/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,135 @@
+	struct timeval now;
+	gettimeofday( &now, NULL );
+	return (double)now.tv_sec + (double)now.tv_usec/1.0e6;
+}
+
+/**
+* Generates a random number on the interval [0,1).
+*
+* @return random number
+*/
+static float rand_float( void ) {
+	int r = rand();
+	return (float)r / ( (float)RAND_MAX + 1.0f );
+}
```
We have moved away from using `rand_float` in our benchmarks. We use this instead:

```c
/**
* Generates a random number on the interval [min,max).
*
* @param min    minimum value (inclusive)
* @param max    maximum value (exclusive)
* @return       random number
*/
static float random_uniform( const float min, const float max ) {
    float v = (float)rand() / ( (float)RAND_MAX + 1.0f );
    return min + ( v*(max-min) );
}
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/acoshf/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,135 @@
+	float y;
+	int i;
+
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = ( 100.0f*rand_float() ) + 1.0f;
+	}
```
These lines will be changed accordingly. Reference: https://github.com/0PrashantYadav0/stdlib/blob/a8785515d4aab7c059e0d1c45e62f656404e5f45/lib/node_modules/%40stdlib/math/base/special/log1pf/benchmark/c/native/benchmark.c#L101-L103

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/acoshf/examples/index.js:
```diff
@@ -0,0 +1,29 @@
+var i;
+for ( i = 0; i < x.length; i++ ) {
+	console.log( 'acoshf( %d ) = %d', x[ i ], acoshf( x[ i ] ) );
+}
```
Can you please use `@stdlib/console/log-each-map` instead? The above coding structure is outdated now.

Reference: https://github.com/0PrashantYadav0/stdlib/blob/a8785515d4aab7c059e0d1c45e62f656404e5f45/lib/node_modules/%40stdlib/math/base/special/log1pf/examples/index.js#L21-L30

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/acoshf/lib/main.js:
```diff
@@ -0,0 +1,122 @@
+
+// VARIABLES //
+
+var HUGE = 1 << 28; // 2**28
```
`var HUGE = f32( 1 << 28 ); // 2**28`

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/acoshf/lib/main.js:
```diff
@@ -0,0 +1,122 @@
+	var t;
+	var s;
+	x = f32( x );
+	if ( isnanf( x ) || x < 1.0 ) {
```
`if ( isnanf( x ) || x < f32( 1.0 ) ) {`

This applies to all the numbers below too. Like 0.0, 1.0, 2.0, etc.

anandkaranubc replied:
Or you can declare them as separate variables. Like https://github.com/0PrashantYadav0/stdlib/blob/a8785515d4aab7c059e0d1c45e62f656404e5f45/lib/node_modules/%40stdlib/math/base/special/log1pf/lib/main.js#L85-L89

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/acoshf/lib/main.js:
```diff
@@ -0,0 +1,122 @@
+* // returns NaN
+*/
+function acoshf( x ) {
+	var t;
+	var s;
+	x = f32( x );
```
```
	var s;

	x = f32( x );
```

anandkaranubc replied:
Leaving a space between the variable declarations and rest of the code makes it a bit cleaner :)

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/acoshf/README.md:
````diff
@@ -0,0 +1,203 @@
+## Examples
+
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var linspace = require( '@stdlib/array/base/linspace' );
+var acoshf = require( '@stdlib/math/base/special/acoshf' );
+
+var x = linspace( 1.0, 5.0, 100 );
+
+var i;
+for ( i = 0; i < x.length; i++ ) {
+    console.log( acoshf( x[ i ] ) );
+}
+```
````
This will be updated accordingly.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/acoshf/README.md:
```diff
@@ -0,0 +1,203 @@
+-   <span class="package-name">[`@stdlib/math/base/special/acos`][@stdlib/math/base/special/acos]</span><span class="delimiter">: </span><span class="description">compute the arccosine of a double-precision floating-point number.</span>
+-   <span class="package-name">[`@stdlib/math/base/special/asinh`][@stdlib/math/base/special/asinh]</span><span class="delimiter">: </span><span class="description">compute the hyperbolic arcsine of a double-precision floating-point number.</span>
+-   <span class="package-name">[`@stdlib/math/base/special/atanh`][@stdlib/math/base/special/atanh]</span><span class="delimiter">: </span><span class="description">compute the hyperbolic arctangent of a double-precision floating-point number.</span>
+
```
You can remove these as these get automatically populated, through our automation scripts.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/acoshf/README.md:
```diff
@@ -0,0 +1,203 @@
+
+[@stdlib/math/base/special/asinh]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/asinh
+
+[@stdlib/math/base/special/atanh]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/atanh
```
Same for these.

nakul-krishnakumar committed:
chore: update benchmark according to code review

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
chore: update according to code review

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
chore: update according to code review

Updated the benchmark for acoshf to use float32 dtype and removed the built-in benchmark.

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
chore: update according to code review

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
chore: update code to meet new conventions

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
chore: update code to meet new conventions

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
chore: update according to code review

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
chore: update according to code review

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
style: remove trailing whitespace

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

anandkaranubc commented:
/stdlib lint-autofix

stdlib-bot committed:
fix: resolve lint errors

anandkaranubc committed:
chore: add line break to fix lint error

Signed-off-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>

nakul-krishnakumar committed:
test: use `ulpdiff` for test comparisons

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
test: use `ulpdiff` for test comparisons

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
test: use `ulpdiff` for test comparisons

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
docs: update README according to code review

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
style: fix lint errors

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
style: fix lint errors

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar commented on lib/node_modules/@stdlib/math/base/special/acoshf/README.md:
```diff
@@ -0,0 +1,192 @@
+
+#### acoshf( x )
+
+Computes the [hyperbolic arccosine][hyperbolic-arccosine] of single-precision floating-point number.
```
```suggestion
Computes the [hyperbolic arccosine][hyperbolic-arccosine] of a single-precision floating-point number.
```

nakul-krishnakumar commented on lib/node_modules/@stdlib/math/base/special/acoshf/README.md:
```diff
@@ -0,0 +1,192 @@
+
+#### stdlib_base_acoshf( x )
+
+Computes the [hyperbolic arccosine][hyperbolic-arccosine] of single-precision floating-point number.
```
```suggestion
Computes the [hyperbolic arccosine][hyperbolic-arccosine] of a single-precision floating-point number.
```

nakul-krishnakumar committed:
docs: update according to code review

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
chore: add `f` suffix for floating point numbers

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

nakul-krishnakumar committed:
chore: add `Float32` conversion

Signed-off-by: Nakul Krishnakumar <nakulkrishnakumar86@gmail.com>

anandkaranubc committed:
docs: update function description

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
bench: add C native benchmarks

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: fix the `len` argument in JS examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
test: update fixtures and use correct `f32` emulation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: update keywords

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: update examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: update copyright years

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc reviewed (dismissed):
Ready to go!

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/acoshf/test/test.js:
```diff
@@ -0,0 +1,139 @@
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.true( typeof acoshf, 'function', 'main export is a function' );
```
Should actually check the assertion here:

```suggestion
	t.strictEqual( typeof acoshf, 'function', 'main export is a function' );
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/acoshf/test/test.native.js:
```diff
@@ -0,0 +1,148 @@
+
+tape( 'main export is a function', opts, function test( t ) {
+	t.ok( true, __filename );
+	t.true( typeof acoshf, 'function', 'main export is a function' );
```
```suggestion
	t.strictEqual( typeof acoshf, 'function', 'main export is a function' );
```

Planeshifter committed:
test: replace t.test with t.strictEqual

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/acoshf/lib/main.js:
````diff
@@ -0,0 +1,126 @@
+* Based on
+*
+* ```tex
+* \operatorname{acoshf}(x) = \logf \left[ x + \sqrtf{ x^2 - 1 } \right]
````
```suggestion
* \operatorname{acoshf}(x) = \log \left[ x + \sqrt{ x^2 - 1 } \right]
```

Planeshifter committed:
chore: fix LaTeX equation

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/acoshf/src/main.c:
````diff
@@ -0,0 +1,97 @@
+* Based on
+*
+* ```tex
+* \operatorname{acoshf}(x) = \logf \left[ x + \sqrtf{ x^2 - 1 } \right]
````
```suggestion
* \operatorname{acoshf}(x) = \log \left[ x + \sqrt{ x^2 - 1 } \right]
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/acoshf/src/main.c:
```diff
@@ -0,0 +1,97 @@
+static const float HUGE = 1 << 28; // 2**28
+
+/**
+*  Computes the hyperbolic arccosine of single-precision floating-point number.
```
```suggestion
* Computes the hyperbolic arccosine of a single-precision floating-point number.
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/acoshf/src/main.c:
````diff
@@ -0,0 +1,97 @@
+* we have
+*
+* ```tex
+* \operatorname{acoshf}(x) = \begin{cases}
+* \logf(x) + \tfrac{\lnf}{2} & \text{ if x is large } \\
+* \logf \left( 2x-\tfrac{1}{\sqrtf{x^2-1}+x} \right) & \text{ if } x > 2 \\
+* \operatorname{log1pf}\left( x - 1 + \sqrtf{ 2 \cdotf (x-1) + (x-1)^2 } \right) & \text{ otherwise }
+* \end{cases}
````
```suggestion
* \operatorname{acoshf}(x) = \begin{cases}
* \log(x) + \tfrac{\ln}{2} & \text{ if x is large } \\
* \log \left( 2x-\tfrac{1}{\sqrt{x^2-1}+x} \right) & \text{ if } x > 2 \\
* \operatorname{log1p}\left( x - 1 + \sqrt{ 2 \cdot (x-1) + (x-1)^2 } \right) & \text{ otherwise }
* \end{cases}
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/acoshf/lib/main.js:
````diff
@@ -0,0 +1,126 @@
+* we have
+*
+* ```tex
+* \operatorname{acoshf}(x) = \begin{cases}
+* \logf(x) + \tfrac{\ln}{2} & \text{ if x is large } \\
+* \logf \left( 2x-\tfrac{1}{\sqrtf{x^2-1}+x} \right) & \text{ if } x > 2 \\
+* \operatorname{log1pf}\left( x - 1 + \sqrtf{ 2 \cdot (x-1) + (x-1)^2 } \right) & \text{ otherwise }
+* \end{cases}
````
```suggestion
* \operatorname{acoshf}(x) = \begin{cases}
* \log(x) + \tfrac{\ln}{2} & \text{ if x is large } \\
* \log \left( 2x-\tfrac{1}{\sqrtf{x^2-1}+x} \right) & \text{ if } x > 2 \\
* \operatorname{log1p}\left( x - 1 + \sqrt{ 2 \cdot (x-1) + (x-1)^2 } \right) & \text{ otherwise }
* \end{cases}
```

Planeshifter committed:
chore: minor clean-up

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/acoshf/include/stdlib/math/base/special/acoshf.h:
```diff
@@ -0,0 +1,38 @@
+#endif
+
+/**
+* Computes the hyperbolic arccosine of single-precision floating-point number.
```
```suggestion
* Computes the hyperbolic arccosine of a single-precision floating-point number.
```

Planeshifter committed:
docs: update description

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
